Client/core/CDiscordManager.cpp

```diff
@@ -0,0 +1,345 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto
+ *  LICENSE:     See LICENSE in the top level directory
+ *  FILE:        Client/core/CDiscordManager.cpp
+ *
+ *  Multi Theft Auto is available from http://www.multitheftauto.com/
+ *
+ *****************************************************************************/
+
+#include "StdInc.h"
+#include "SharedUtil.Thread.h"
+#include <chrono>
+
+CDiscordManager::CDiscordManager() : m_discordCore(nullptr), m_bSuicide(false), m_waitingForServerName(false), m_storedActivity{}
+{
+    Reconnect(true); // Try to interact with discord on construction
+    m_pThread = new CThreadHandle(CDiscordManager::DiscordThread, this);
+
+    m_storedActivity.GetAssets().SetLargeImage("multi-theft-auto"); // Always thing
+    m_storedActivity.GetAssets().SetLargeText("Playing MTA:SA");
+}
+
+CDiscordManager::~CDiscordManager()
+{
+    m_bSuicide = true;
```

Comment:
This look very hacky. Not sure how to do this more nicely though

Reply:
The only other way is signals I guess